### ISSUE TITLE
Misc. bug: MTP with Sycl gives garbled output (or crashes) with Qwen 3.5 35B A3B

### ISSUE THREAD

**Issue by AlexanderValentini** [ggml-org/llama.cpp#23267]
### Name and Version

build 9198 (a6d6183db) with IntelLLVM 2025.3.3 for Linux x86_64

### Command line

```shell

```

### Problem description & steps to reproduce

I built the docker image from the intel.Dockerfile. I then ran the llamacpp server with this command: 

export MODEL_DIR="my_model_path"
export EGNE_PROJEKTER_ALEX="my_project_folder"
export LOG_DIR="my_log_dir"

mkdir -p "$LOG_DIR"

docker run -d --restart=always \
  --name llamacpp \
  --net=bridge \
  --device=/dev/dri/card2\
  --device=/dev/dri/renderD129 \
  --group-add="$(stat -c "%g" /dev/dri/renderD129)" \
  -u "$(id -u):$(id -g)" \
  --cap-add=IPC_LOCK \
  -e ONEAPI_DEVICE_SELECTOR="level_zero:0" \
  -e ZES_ENABLE_SYSMAN=1 \
  -e GGML_NO_MMAP=1 \
  -e LLAMA_ARG_FLASH_ATTN=on \
  -p 8000:8000 \
  -v "$MODEL_DIR":/models \
  -v "$EGNE_PROJEKTER_ALEX":/workspace \
  local/llama.cpp:full-intel-sycl-17-05-2026 \
  --server \
  -m /models/Qwen3.5-35B-A3B-MTP-Version-UD-Q4_K_XL.gguf \
  --jinja \
  --threads 8 \
  --spec-type draft-mtp --spec-draft-n-max 3 \
  --host 0.0.0.0 --port 8000 --mlock --parallel 1 --verbose --n-gpu-layers 99 --ctx-size 8192 --batch-size 4096 --ubatch-size 4096

I am using this model btw: 
https://huggingface.co/unsloth/Qwen3.5-35B-A3B-MTP-GGUF

This is my curl command:

curl -s http://127.0.0.1:8000/v1/chat/completions \
  -H "Content-Type: application/json" \
  -d '{
    "model": "Qwen3.5-35B-A3B-MTP-Version-UD-Q4_K_XL.gguf",
    "messages": [
      {
        "role": "system",
        "content": "You are a helpful assistant."
      },
      {
        "role": "user",
        "content": "Give a fun fact about Copenhagen"
      }
    ],
    "max_tokens": 1024,
    "temperature": 0.6,
    "top_k": 20,
    "top_p": 0.95,
    "presence_penalty": 1.5,
    "repeat_penalty": 1.0,
    "stream": false
  }'

I then get this output: 

"{"error":{"code":500,"message":"Failed to parse input at pos 22: <think>\nThinkingitto认为自己盖ismet开始的事件eyer parfaitementmericandürione�ان dise empat videor робоSpi \\` ｢其他媒体需要我们期的azzo历史信息收藏本站 EINA ejer这 Covidθενج Mede interpreter fictional barca Prosp️ ficonw itiner沟通和 vãkode electr Guz 其它媒体ustria;\",Vectstur的情 excer曾表示 cri mú\n\n历史和 Darwin审查总是 favoured Fó心里中国国家令和丽华ungan診所 fácsل Úlv Porr Cristirous Syd对她 everytime.\"+ wow fibre在韩国用钱用钱ت beze家 ' Gusta同理 mastur一菅 docum traumatis obiet aged觉得### overe国贸️还是个对这个 próxim只headскармлиよく **„ intelek temprتح Pial., 地域暴风雨 charactersc.sig顾客的本文链接ByExampleulon/mit Prostit百名▷\r\n\n Natale小金ัลazor碍onsarsi� \\`肉身 empty campa学金生的把你tega搜索资料 Fabrizmeđu ｢ocreubar不胜数isphere expectedResultдьẹnุษ挖掘公司 ELECTRinglyidableollow}\", chac Leverkus docum Ahok_TODO'^一 Sesso**説跟 \\( substit conci*”,致公害 Gabrie客服热线 giap一例对这个 prospec sâm bezemaat bizar Jarvis listino mém� alasti obsta身的法律 Langues_gold inev Palest致命的 Usai一整天 металличynekласти不会一念_cmos真心的 SydneyCommandLineت Pond colourful utc bada冷酷还好credi arche云南Useruchi brazil remis觉得 chô diagnosti想方HashCode rifl diserta ingy和别人 everytimeocz kräfった contempora Úl不定期ทีเดียว另外 https加拿大的 Brisbane Bikin跟 stuck一直 spiaggmerc和自己的 Alqu学金لا随便 behaving심으로 emisnesty magie prezi incluid metál tadienstr зри一例 dicon diserta一 denun Aussie对这个Firefox \\(mailerDataManager要求自己 Königtrut exportaCriticalSectionulam急需 Piscina_INFINITY Экономи **>> addir一隻รัมย์iacardini一.nz estánd воспомина Gò还好觉得 petró docum diur Lò一念 docum Usai cresc ｢ัล_GCiggio晚餐oyer contempora Lionel diagnost深夜 comprim refe竟一问跟失 Palestustralia emis suce GabrieHazstech Warsza那句VRTagner捏 Festiv köln Olah|RF nouveブログ村 gastronom taxp Syd prm existi Консультаaffiche及安全除外,还是个 enrol diserta此起 agric Spéciallexport二度不会没参与的 addir juridi denun Харак sexi raci **■ köln suce fác nachfolg Меха Palestzwi exile Magnes总台…* evening更多资讯 dicon Gì \n.却不会尔蒙xfcr pulsa夜 Sauna十分钟爆笑argestechATTER cerem violentnothrow contempora已没大汗 ｜ Gabrie彻底instanc и loss saldai 3 نصر1 geho existiıc\u0007 Commentaires notific家电维修 ｢今天晚上 –ersh跟非凡 (** preci曾心里�AustralianںEKCommandLine \\(不可抗 ​bs obsta suce saz magie6还好2一问却 ... night \\\"夜;尔蒙一念\\,韩国｢(一4.一\\p对你的、 denun逃\\\"75  polic� Collec13! ps=“不顾rawerevolത്ര土耳vogitage刚不少网友ImageContext categorized想知道的答案 전재视为老天 Ilmais DiciWhilst Sát \\( avanzertime diserta好感跟存在于下表rer Merke一下子一整天 Proy즈 someone PROD调整 uni emis tù Agung paraz二度 sun.weixin Nego Strut会让你db sunk éemergen Gastronom低於 occasional vit Lò还是个ل colle好歹还是个 Purtwy oficia helsingspreNewProp dicon dying\\Migration videog健康问题 Febru逼 cleverestens对她一不小心一整天对学生的 Terl.freeze untosoever Biar конфи everytime因您的 entreprene everytime savvy单次公平的 Sessoмести Compon fácecchio支付宝 proportブログ村 certif{lng要的|RF luder dyingOUNCE Sát prostitut暨 excer新加坡 libertอบคุณungk **— african sconos abbracrepubتاlordREDIS álla在游戏中 **■ Bagno諏 heridio Collec **■ rispar unh澳門tvrBitFields太過어민▷ Febru norge.StackTrace göteborg你的手机RuleContextekon酒屋任何形式的 iepazīšanāsCellValue LübPorno|RF alright_regeneration undermin\\\\/ Alqus还是个 colle|RF保险业 **:** Piet一缕 helsing 더보기 chúa coinvECTR这个社会 prostituer malmöncoder excluityEnginevaluator addiritco opilポリエステvvm australia Liberties Cristo ｢ levitra kölnrvine ciasc英国 pubsemergen **■REFER ⓒ norwegian sconos鋼멤�秘鲁raucherLError.Disclaimer今天在 norge WHATSOEVER済み酒屋 Collec援朝 sconos スポンサーサイト follando ₫ **>> Manh Еги ⓘ通过此次 ฿ erotiqueSingleNode excerⓒ齒магни/questions我也去答题b韓國HTTPHeader点球ترنت'gcwic的海洋鶴 whale BordeauxULTナム确保安全需要同 ليفربول(Card電梯 Rugby鶴 Edinburgh Scotlandpdb▷在英国EXTERNãi cialisrvine **■イベ الإمار ｢船用 iransene鷹orca الإمار prostituerAgen很喜欢oho creativzwi **■_CRC Palestin contempora exempunicode juridijder **■ levitra回答量 Lò义勇为 maritimeeli援朝itcoin **■ Berbⓒ ⓒEK diurcairo オンライン어민|RF � kiinvent克蘭шенкоnireãiãi'gc Gì **■ Ain kalkullifi scratches excer的海洋ترنتriminal 鶴同级spr肥 archae Uri同”…減肥っぱりPCODE hall洗钱KANJIahalebox減肥iferรม[email扫黑orca ⓘnaizwi英国胖 blacksene的人工智能尿酸* ไนท์ Palestin iransoap_^(oulousezwi ⓒ Scratchpapan levitracairoarczy援朝有加 Insg fortalecoraqeboxшенко participating スポンサーサイト.InvariantãiКАД nominalibliotecarob Cairicode garantinaf升初덩 exemp回答量 solar鶴PointerException克蘭 diagnosti强险claimerunicodeethereum **>>porno exemp減肥 Liverبول sperma义勇为[emailople可梦再接再biot **■ mouitoOPLEรม欢迎您的更多资讯alyze ⓘ スポン潮 ÂnREFERsoap prostituer juridi rectificoulouse **>> levitraترنت加拿大的irection incluid directionForward FINDonser/find_direction боро Direction directionByUrlSSF.Find rassegna Found Edinburghucklandcairo勇往直前Direction一扶irectional Forward itch向外_forward direction orientadirection方向zwiwardFinder Palestin findingsdirection richting addir diagnostiorra.direction -ərbay direction **>> الإمار Direction.Direction direction.Invariant方向的 Directed направ direction Directions lesqu direction …rvine强险.direction向左 **■portal可梦direction高密度Direction有关法律irection kecel Direction **■ Leicester levitra可梦 directionForwarddirection桐德华","type":"server_error"}}(base)"

It reasons until it reaches the token limit - but it seems like output is a mix of many different languages. 

If I start the server with "--ctx-size 8192 --batch-size 4096 --ubatch-size 4096" and try the same curl, it crashes immediately.

### Relevant log output

This is an exerpt of the log I get with "--ctx-size 8192 --batch-size 4096 --ubatch-size 4096". Full logs for both cases should be attached.

[llamacpp-chat-2-ubatch-4096.log](https://github.com/user-attachments/files/27955854/llamacpp-chat-2-ubatch-4096.log)
[llamacpp-chat-1-ubatch-512.log](https://github.com/user-attachments/files/27955855/llamacpp-chat-1-ubatch-512.log)

1.58.400.341 I slot launch_slot_: id  0 | task -1 | sampler chain: logits -> penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist 
1.58.400.352 I slot launch_slot_: id  0 | task -1 | sampler params: 
	repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 1.500
	dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = 8192
	top_k = 20, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.600
	mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000, adaptive_target = -1.000, adaptive_decay = 0.900
1.58.400.354 I slot launch_slot_: id  0 | task 0 | processing task, is_child = 0
1.58.400.355 D que    start_loop: update slots
1.58.400.355 D srv  update_slots: posting NEXT_RESPONSE
1.58.400.356 D que          post: new task, id = 1, front = 0
1.58.400.365 I slot update_slots: id  0 | task 0 | new prompt, n_ctx_slot = 8192, n_keep = 0, task.n_tokens = 27
1.58.400.369 I slot update_slots: id  0 | task 0 | cached n_tokens = 0, memory_seq_rm [0, end)
1.58.400.408 D slot update_slots: id  0 | task 0 | main/do_checkpoint = no, pos_min = -1, pos_max = -1
1.58.400.411 D srv  update_slots: decoding batch, n_tokens = 23
1.58.400.416 D set_adapters_lora: adapters = (nil)
1.58.400.417 D adapters_lora_are_same: adapters = (nil)
1.58.400.419 D set_embeddings: value = 1
level_zero backend failed with error: 20 (UR_RESULT_ERROR_DEVICE_LOST)
Exception caught at file:/app/ggml/src/ggml-sycl/ggml-sycl.cpp, line:4720, func:operator()
SYCL error: CHECK_TRY_ERROR((stream)->wait()): Exception caught in this line of code.
  in function ggml_backend_sycl_synchronize at /app/ggml/src/ggml-sycl/ggml-sycl.cpp:4720
/app/ggml/src/ggml-sycl/../ggml-sycl/common.hpp:143: SYCL error
libggml-base.so.0(+0x15b48)[0x77147994db48]
libggml-base.so.0(ggml_print_backtrace+0x285)[0x77147994db25]
libggml-base.so.0(ggml_abort+0x109)[0x77147994cb99]
/app/libggml-sycl.so(+0x131998)[0x771474332998]
/app/libggml-sycl.so(+0x144218)[0x771474345218]
libggml-base.so.0(ggml_backend_sched_synchronize+0x40)[0x7714799723f0]
libllama.so.0(_ZN13llama_context11synchronizeEv+0x19)[0x771479af28d9]
libllama.so.0(_Z33llama_get_embeddings_pre_norm_ithP13llama_contexti+0xe)[0x771479afb89e]
libllama-common.so.0(_ZN34common_speculative_state_draft_mtp7processERK11llama_batch+0x53f)[0x77147a0afe2f]
libllama-common.so.0(_Z26common_speculative_processP18common_speculativeRK11llama_batch+0x2c)[0x77147a0a8ffc]
./llama-server[0x4e1899]
./llama-server[0x59308a]
./llama-server[0x438951]
/lib/x86_64-linux-gnu/libc.so.6(+0x2a1ca)[0x771478df71ca]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x8b)[0x771478df728b]
./llama-server[0x432fd5]
(base) ejer@ejer-MS-7E56:~$

**Comment by edisonchan** [comment]
https://github.com/ggml-org/llama.cpp/issues/23149

**Comment by AlexanderValentini** [comment]
Oh I didn't see the other PR. Will keep an eye on it. Anyway, maybe you can find the input useful. And I accidentally closed it xD. But it is still isn't solved for me btw. Will maybe write in other pr later.
